packages/components/src/ListBox/src/ListBoxItem.tsx

```diff
@@ -0,0 +1,313 @@
+    /**
+     * Whether or not the ListBoxItem is in an invalid state.
+     */
+    isInvalid?: boolean;
+    /**
+     * The selection indicator to use. Only available if the selection mode is not "none".
+     * When set to "input", the selection indicator will be an either a radio or checkbox based on the selection mode.
+     * @default "check"
+     */
+    selectionIndicator?: "check" | "input";
+    /**
+     * A ListBoxItem can vary in size.
+     * @default "sm"
+     */
+    size?: ResponsiveProp<ListBoxItemSize>;
+}
+
+interface ListBoxInnerProps extends ListBoxItemRenderProps {
+    /**
+     * The selection indicator to use. Only available if the selection mode is not "none".
+     * When set to "input", the selection indicator will be an either a radio or checkbox based on the selection mode.
+     * @default "check"
+     */
+    selectionIndicator?: "check" | "input";
+    /**
+     * Whether or not the ListBoxItem is in an invalid state.
+     */
+    isInvalid?: boolean;
+    /**
+     * A ListBoxItem can vary in size.
+     * @default "sm"
+     */
+    size: ListBoxItemSize;
+    /**
+     * The children of the ListBoxItem.
+     */
+    children: ReactNode;
+    /**
+     * A function to set the list has selection state.
+     */
+    setListHasSelection: (value: boolean) => void;
+}
+
+const ListBoxItemToIconSizeAdapter = {
+    xs: "sm",
+    sm: "sm",
+    md: "md",
+    lg: "md"
+} as const satisfies SizeAdapter<ListBoxItemSize, IconProps["size"]>;
+
+const ListBoxItemToTextSizeAdapter = {
+    xs: "sm",
+    sm: "sm",
+    md: "sm",
+    lg: "md"
+} as const satisfies SizeAdapter<ListBoxItemSize, TextProps["size"]>;
+
+const getIsListHasSelectionEnd = (isListHasSelection: boolean, prevIsListHasSelectionEnd: boolean): boolean => {
+    return isListHasSelection && !prevIsListHasSelectionEnd;
+};
+
+function ListBoxItemInner(props: ListBoxInnerProps) {
+    const listStateContext = useContext(ListStateContext);
+    
+    const { selectionMode, 
+        isDisabled, 
+        isFocusVisible, 
+        isPressed, 
+        isHovered, 
+        isSelected, 
+        selectionIndicator, 
+        isInvalid, 
+        size, 
+        children,
+        setListHasSelection
+    } = props;
+    
+    const isRadio = selectionIndicator === "input" && selectionMode === "single";
+    const isCheckbox = selectionIndicator === "input" && selectionMode === "multiple";
+    const isCheck = selectionIndicator === "check" && selectionMode !== "none";
+
+    useEffect(() => {
+        setListHasSelection(listStateContext.selectionManager.selectedKeys.size > 0);
+    }, [listStateContext.selectionManager.selectedKeys.size, setListHasSelection]);
+
+    return (
+        <>
+            {isRadio && (
+                <RadioGroup 
+                    size="sm"
+                    aria-hidden="true" 
+                    aria-label="list item indicator"
```

Comment:
did you add this because there was a warning? try setting an empty string. i don't like that there is an hardcoded string in a language that might not be the right one

Reply:
I'll try, because yes there was a warning. I could add a translation is we need it.